contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -123,22 +206,24 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
                 : startDate + fetchOfferDurations(_offerId).voucherValid;
 
             // Map the offerId to the exchangeId as one-to-many
-            lookups.exchangeIdsByOffer[_offerId].push(exchangeId);
+            lookups.exchangeIdsByOffer[_offerId].push(_exchangeId);
 
-            // Shouldn't decrement if offer is unlimited
-            if (offer.quantityAvailable != type(uint256).max) {
```

Comment:
i guess, re: unlimited, i wouldn't have done a special case here, where we treat uint256.max as special, i would have just decremented uint256.max that that is pretty much infinity for all practical purposes, removing more code...

Reply:
Removing code here, adds code elsewhere - dapp would need to decide then when they show that offer is unlimited and when they show actual value they got when they query getOffer. If we maintain uint256.max then it's unambiguous.

We also have some other places where unlimited offers are treated differently and it's good to have a reliable identifier.

Reply:
:)